web/proxy.js

```diff
@@ -59,6 +59,16 @@ const server = http.createServer((request, response) => {
 
     request.pipe(proxyRequest);
     proxyRequest.on('response', (proxyResponse) => {
+        // Get the existing CSP header value
+        let existingCSP = proxyResponse.headers['content-security-policy'];
+
+        // Append newDot staging and development URLs in the frame-ancestors CSP headers to allow the oldDot to be embedded in the Expensify App
+        existingCSP += ' https://staging.new.expensify.com http://localhost:8082';
```

Comment:
This assumes we never change the policy format and `frame-ancestors` is always the last property returned in the header. Is there a safer way to do this?

Reply:
I actually wonder if we need this PR at all, considering [our new approach of](https://expensify.slack.com/archives/C05NM77EUTS/p1694684405171459?thread_ts=1694684392.189589&cid=C05NM77EUTS) configuring NewDot Cloudflare proxy to serve OldDot pages.

If my understanding is correct, with the new approach, this is the flow that happens on dev? @techievivek

- If on dev and we want to display, for example, the OldDot Inbox page inside an iframe, we set the `src` attribute of iframe to `localhost:8082/olddot/inbox`  or just `/olddot/inbox`
- The dev server running on the port `8082` gets the request
- Because we will add the mapping of `olddot` -> `new.expensify.com/olddot` [here](https://github.com/Expensify/App/blob/c511efdfb75cf41a5410311e14350e28370dd3cb/config/webpack/webpack.dev.js#L23-L28), the request is forward to `new.expensify.com`
- We will configure the NewDot Cloudflare proxy to forward `new.expensify.com/olddot` -> `expensify.com`
- The response comes back like this: `expensify.com` -> Cloudflare proxy -> dev server -> iframe inside dev NewDot app

So from the browser perspective, the Inbox page is served from `localhost:8082`, thus the Same Origin policy will apply. Even without specifying `localhost:8082`, browser can display the Inbox page inside the iframe.

Reply:
> Because we will add the mapping of olddot -> new.expensify.com/olddot [here](https://github.com/Expensify/App/blob/c511efdfb75cf41a5410311e14350e28370dd3cb/config/webpack/webpack.dev.js#L23-L28), the request is forward to new.expensify.com

With this config, I think it should work, yes.